bigtable/client/internal/table_admin.h

```diff
@@ -68,7 +75,37 @@ class TableAdmin {
         instance_name_(InstanceName()),
         rpc_retry_policy_(retry_policy.clone()),
         rpc_backoff_policy_(backoff_policy.clone()),
-        metadata_update_policy_(instance_name(), MetadataParamTypes::PARENT) {}
+        metadata_update_policy_(instance_name(), MetadataParamTypes::PARENT),
+        polling_policy_(DefaultPollingPolicy()) {}
+
+  /**
+   * Create a new TableAdmin using explicit policies to handle RPC errors.
+   *
+   * @tparam RPCRetryPolicy control which operations to retry and for how long.
+   * @tparam RPCBackoffPolicy control how does the client backs off after an RPC
+   *     error.
+   * @tparam PollingPolicy provides parameters for asynchronous calls.
+   * @param client the interface to create grpc stubs, report errors, etc.
+   * @param instance_id the id of the instance, e.g., "my-instance", the full
+   *   name (e.g. '/projects/my-project/instances/my-instance') is built using
+   *   the project id in the @p client parameter.
+   * @param retry_policy the policy to handle RPC errors.
+   * @param backoff_policy the policy to control backoff after an error.
+   * @param polling_policy the policy to control the asynchronous call
+   * parameters
+   */
+  template <typename RPCRetryPolicy, typename RPCBackoffPolicy,
+            typename PollingPolicy>
+  TableAdmin(std::shared_ptr<AdminClient> client, std::string instance_id,
+             RPCRetryPolicy retry_policy, RPCBackoffPolicy backoff_policy,
+             PollingPolicy polling_policy)
+      : client_(std::move(client)),
+        instance_id_(std::move(instance_id)),
+        instance_name_(InstanceName()),
+        rpc_retry_policy_(retry_policy.clone()),
+        rpc_backoff_policy_(backoff_policy.clone()),
+        metadata_update_policy_(instance_name(), MetadataParamTypes::PARENT),
+        polling_policy_(std::move(polling_policy)) {}
```

Comment:
The polling policy is different from the other policies in that they are received by value and polling is received as a `unique_ptr<>`.   Can you fix this?  If not, can you open a bug and put a TODO here?

Reply:
Shall I have to receive PollingPolicy by value? Same as other policies.

Reply:
Yes please.